app/src/androidTest/java/com/github/sdp/mediato/DatabaseTests.java

```diff
@@ -5,52 +5,143 @@
 import static org.junit.Assert.assertThrows;
 import static org.junit.Assert.assertTrue;
 
+import androidx.annotation.NonNull;
 import androidx.test.ext.junit.runners.AndroidJUnit4;
 
 import com.github.sdp.mediato.data.Database;
 import com.github.sdp.mediato.model.Location;
 import com.github.sdp.mediato.model.User;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.ValueEventListener;
 import com.google.firebase.firestore.FirebaseFirestore;
 
-import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
+
 @RunWith(AndroidJUnit4.class)
+
 /**
  * This class contains all the tests for database interactions
+ *
  * @TODO add the Cloud Storage tests for the profile pictures
  */
 public class DatabaseTests {
     private static int STANDARD_TIMEOUT = 10;
     FirebaseFirestore db = FirebaseFirestore.getInstance();
     User user1;
+    User user2;
+    User user3;
 
     @Before
-    public void setUp(){
-       try {
-           Database.database.useEmulator("10.0.2.2", 9000);
+    public void setUp() {
+        try {
+            Database.database.useEmulator("10.0.2.2", 9000);
+        } catch (Exception e) {
         }
-        catch(Exception e){}
-        //Create new sample user
+        //Create new sample users
         user1 = new User.UserBuilder("uniqueId1")
                 .setUsername("user_test_1")
                 .setEmail("email_test_1")
                 .setRegisterDate("09/03/2023")
                 .setLocation(new Location(3.14, 3.14))
                 .build();
+        user2 = new User.UserBuilder("uniqueId2")
+                .setUsername("user_test_2")
+                .setEmail("email_test_2")
+                .setRegisterDate("19/03/2023")
+                .setLocation(new Location(3.14, 3.14))
+                .build();
+        user3 = new User.UserBuilder("uniqueId3")
+                .setUsername("user_test_3")
+                .setEmail("email_test_3")
+                .setRegisterDate("19/03/2023")
+                .setLocation(new Location(3.14, 3.14))
+                .build();
     }
 
-    @After
-    //Cleans database values after each test
-    public void cleanDatabase(){

```

Comment:
why was this removed ? It seems important to me that the emulator database is cleaned after tests complete

Reply:
It was breaking some tests because they don't wait for each other to complete but now I changed it to @AfterClass that way it cleans the database after all the tests are executed instead.